components/movementsensor/adxl345/adxl345.go

```diff
@@ -121,22 +148,41 @@ func NewAdxl345(
 	if !ok {
 		return nil, errors.Errorf("can't find I2C bus '%q' for ADXL345 sensor", cfg.I2cBus)
 	}
-
 	var address byte
 	if cfg.UseAlternateI2CAddress {
 		address = 0x1D
 	} else {
 		address = 0x53
 	}
-
+	interruptConfigurations := getInterruptConfigurations(cfg)
+	configuredRegisterValues := map[byte]byte{}
+	configuredRegisterValues = getFreeFallRegisterValues(cfg.FreeFall, configuredRegisterValues)
```

Comment:
re. getFreeFallRegisterValues. this is an odd pattern of passing a map, which get's modified and returned.
can you have either 
1. have `getFreeFallRegisterValues` return a map of new values that need to be added, and merge the 2 maps or
2. pass configRegisterValues as a pointer and add values in the function